### PR TITLE
docs: templates and descriptions

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
This PR adds a template for the resource and data resource `spectrocloud_application_profile.` Additionally, descriptions of each attribute are also provided. 

### Data Source:
CleanShot 2022-11-29 at 11 51 48

### Resource
**Note**: Ignore the artifact from the scrolling screenshot 😓 

CleanShot 2022-11-29 at 11 53 05

### CODE REVIEW

File: docs/resources/application_profile.md [hunk 1]
```diff
@@ -1,33 +1,149 @@
...
+- `pack` (Block List, Min: 1) A list of packs to be applied to the application profile. (see [below for nested schema](#nestedblock--pack))
 
 ### Optional
 
-- `cloud` (String)
-- `context` (String)
-- `description` (String)
-- `tags` (Set of String)
+- `cloud` (String) The cloud provider the app profile is eligble for usage. Default value is `all`.
+- `context` (String) Context of the application profile. Allowed values are `project`, `cluster`, or `namespace`. Default value is `project`.
```
Comment (nikchern) [inline docs/resources/application_profile.md:142]: allowed values are project cluster and system. namespace is not allowed.

**Review by nikchern — CHANGES_REQUESTED** [review]

File: docs/resources/application_profile.md (hunk 1 above)
Comment (karl-cardenas-coding) [inline docs/resources/application_profile.md:142]:
```suggestion
- `context` (String) Context of the application profile. Allowed values are `project`, `cluster`, or `system`. Default value is `project`.
```

Comment (karl-cardenas-coding) [reply docs/resources/application_profile.md]: @nikchern fixed, nice catch 😄

**Comment by nikchern** [comment]
@TylerGillson 
@karl-cardenas-coding 
I was looking through the documentation that you are writing for TF. I researched one approach where we can reuse schema
So there is no need to duplicate documentation for different cloud types for the most part.
Let's not merge this change and use it as an opportunity to try to reuse schema.

**Comment by karl-cardenas-coding** [comment]
@nikchern what would the new approach look like? We will still have the need to have custom content for each resource, primarily for the usage example section.

File: spectrocloud/resource_application_profile.go [hunk 2]
```diff
@@ -28,84 +29,101 @@ func resourceApplicationProfile() *schema.Resource {
...
+				Optional:    true,
 			},
 			"cloud": {
-				Type:     schema.TypeString,
-				Default:  "all",
-				Optional: true,
-				ForceNew: true,
+				Type:        schema.TypeString,
+				Default:     "all",
+				Description: "The cloud provider the app profile is eligble for usage. Default value is `all`.",
```
Comment (nikchern) [inline spectrocloud/resource_application_profile.go:70]:
```suggestion
				Description: "The cloud provider the profile is eligible for. Default value is `all`.",
```

File: spectrocloud/resource_application_profile.go [hunk 3]
```diff
@@ -28,84 +29,101 @@ func resourceApplicationProfile() *schema.Resource {
...
+				Description: "A list of tags to be applied to the application profile. Tags must be in the form of `key:value`.",
 				Elem: &schema.Schema{
 					Type: schema.TypeString,
 				},
 			},
 			"description": {
-				Type:     schema.TypeString,
-				Optional: true,
+				Type:        schema.TypeString,
+				Description: "Description of the application profile.",
```
Comment (nikchern) [inline spectrocloud/resource_application_profile.go:64]:
```suggestion
				Description: "Description of the profile.",
```

File: spectrocloud/resource_application_profile.go [hunk 4]
```diff
@@ -28,84 +29,101 @@ func resourceApplicationProfile() *schema.Resource {
...
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "1.0.0", // default as in UI
+				Description: "Version of the application profile. Default value is 1.0.0.",
 			},
 			"context": {
 				Type:         schema.TypeString,
 				Optional:     true,
 				Default:      "project",
+				Description:  "Context of the application profile. Allowed values are `project`, `cluster`, or `namespace`. Default value is `project`.",
```
Comment (nikchern) [inline spectrocloud/resource_application_profile.go:50]:
```suggestion
				Description:  "Context of the profile. Allowed values are `project`, `cluster`, or `namespace`. Default value is `project`.",
```

File: spectrocloud/resource_application_profile.go [hunk 5]
```diff
@@ -28,84 +29,101 @@ func resourceApplicationProfile() *schema.Resource {
...
+				Required:    true,
 			},
 			"version": {
-				Type:     schema.TypeString,
-				Optional: true,
-				Default:  "1.0.0", // default as in UI
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "1.0.0", // default as in UI
+				Description: "Version of the application profile. Default value is 1.0.0.",
```
Comment (nikchern) [inline spectrocloud/resource_application_profile.go:44]:
```suggestion
				Description: "Version of the profile. Default value is 1.0.0.",
```

**Pull request merged**